pkg/alertmanager/api_grafana.go

```diff
@@ -419,3 +430,62 @@ func (am *MultitenantAlertmanager) DeleteUserGrafanaConfig(w http.ResponseWriter
 	w.WriteHeader(http.StatusOK)
 	util.WriteJSONResponse(w, successResult{Status: statusSuccess})
 }
+
+// ValidateUserGrafanaConfig validates the Grafana Alertmanager configuration.
+func validateUserGrafanaConfig(logger log.Logger, cfg alertspb.GrafanaAlertConfigDesc, limits Limits, user string) error {
+	// We don't have a valid use case for empty configurations. If a tenant does not have a
+	// configuration set and issue a request to the Alertmanager, we'll a) upload an empty
+	// config and b) immediately start an Alertmanager instance for them if a fallback
+	// configuration is provisioned.
```

Comment:
Nit: Something along the lines of:
```suggestion
	// configuration set and issue a request to the Alertmanager, we'll a) upload an empty
	// config and b) on the next configuration sync, start an Alertmanager instance for them if a fallback
	// configuration is provisioned.
```

Or omit the "immediately" part. It's not true now, but it will be if we implement _strict initialization_.